### PR TITLE
[8.x] fix doc blocks

### PR DESCRIPTION
**Pull request by rodrigopedra**
Addresses errors for 8.x branch listed on #35673 .

**EDIT**: I also made some other fixes I found while assessing doctum error messages.

To clarify, I didn't run doctum myself. I based my code review on the error messages listed on #35673

### CODE REVIEW

File: src/Illuminate/Mail/PendingMail.php
Comment (rodrigopedra) [inline src/Illuminate/Mail/PendingMail.php:117]: Here I removed the extra blank line between parameters and return, and also changed the return to match the interface.

Mailer implementation `send` method does not return anything.

File: src/Illuminate/Notifications/NotificationSender.php
Comment (rodrigopedra) [inline src/Illuminate/Notifications/NotificationSender.php:174]: The implementation does not expects an array on this parameter

File: src/Illuminate/Support/Testing/Fakes/MailFake.php
Comment (rodrigopedra) [inline src/Illuminate/Support/Testing/Fakes/MailFake.php:350]: Changed to match its interface

File: src/Illuminate/Support/Testing/Fakes/PendingMailFake.php
Comment (rodrigopedra) [inline src/Illuminate/Support/Testing/Fakes/PendingMailFake.php:25]: Here I removed the extra semicolon, and also changed the return to match the interface.

Mailer implementation send method does not return anything.

File: src/Illuminate/Mail/PendingMail.php
Comment (williamdes) [reply src/Illuminate/Mail/PendingMail.php:117]: Oh okay, did Doctum say something about that ?

**Comment by rodrigopedra** [comment]
From the errors for the 8.x branch the only I left unaddressed was this:

~~~text
ERROR: "0" @param tags are expected but only "1" found on "Illuminate\Http\Client\Response::throw" in build/doctum/laravel/src/Illuminate/Http/Client/Response.php:236
~~~

This was a feature added by commit https://github.com/laravel/framework/commit/d034e2c55c6502fa0c2bebb6cbf99c5e685beaa5 to allow a callback but avoid a breaking change on changing the method's signature.

Maybe the method signature could be changed for 9.x

**Comment by williamdes** [comment]
> From the errors for the 8.x branch the only I left unaddressed was this:
> 
> ```
> ERROR: "0" @param tags are expected but only "1" found on "Illuminate\Http\Client\Response::throw" in build/doctum/laravel/src/Illuminate/Http/Client/Response.php:236
> ```
> 
> This was a feature added by commit [d034e2c](https://github.com/laravel/framework/commit/d034e2c55c6502fa0c2bebb6cbf99c5e685beaa5) to allow a callback but avoid a breaking change on changing the method's signature.
> 
> Maybe the method signature could be changed for 9.x

Seems to be a Doctum 5.3.0-dev bug from the source code, I see no reason to remove the phpdoc tag (from an external view)

**Comment by williamdes** [comment]
```
Version 8.x
------------
  Parsing   done

ERROR: The method "cache" has "1" @param tags but only "0" where expected. on "\cache" in /mnt/Dev/@code-lts/@doctum-fork/laravel.com-next/build/doctum/laravel/src/Illuminate/Foundation/helpers.php:233
ERROR: The "args" parameter of the method "event" is missing a @param tag on "\event" in /mnt/Dev/@code-lts/@doctum-fork/laravel.com-next/build/doctum/laravel/src/Illuminate/Foundation/helpers.php:430
ERROR: The method "event" has "3" @param tags but only "1" where expected. on "\event" in /mnt/Dev/@code-lts/@doctum-fork/laravel.com-next/build/doctum/laravel/src/Illuminate/Foundation/helpers.php:430
ERROR: The method "throw" has "1" @param tags but only "0" where expected. on "Illuminate\Http\Client\Response::throw" in /mnt/Dev/@code-lts/@doctum-fork/laravel.com-next/build/doctum/laravel/src/Illuminate/Http/Client/Response.php:236

 988/988 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% 16 secs
  Rendering                                             Rendering done

 1123/1123 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100% 23 secs


```

Here are the errors that are not fixed, thank you for your quick work on the other ones !

**Comment by williamdes** [comment]
> Seems to be a Doctum 5.3.0-dev bug from the source code, I see no reason to remove the phpdoc tag (from an external view)

Nonsense .. :smile: 

At https://github.com/laravel/framework/blob/13462f6cbb07b725f9353808d33a149243e49bdc/src/Illuminate/Http/Client/Response.php#L231 it needs to be removed ^^

File: src/Illuminate/Mail/PendingMail.php
Comment (rodrigopedra) [reply src/Illuminate/Mail/PendingMail.php:117]: No, but when I was reviewing the others I found this. Neither doctum or StyleCI complained about this one.

I will update my opening comment to clarify I made other fixes.

Comment (williamdes) [reply src/Illuminate/Mail/PendingMail.php:117]: phpstan would have got it I think ;p

File: src/Illuminate/Notifications/NotificationSender.php
Comment (rodrigopedra) [reply src/Illuminate/Notifications/NotificationSender.php:174]: Hi @williamdes 

Maybe this one you want to take a look on how doctum is handling, the error message was odd for this line:

~~~text
ERROR: The "r" @param tag variable name is wrong (should be "notification") on "Illuminate\Notifications\NotificationSender::queueNotification" in build/doctum/laravel/src/Illuminate/Notifications/NotificationSender.php:177
~~~

As you can see the `@param` variable was not called `r`. Maybe it is because it used this odd syntax with brackets

Comment (williamdes) [reply src/Illuminate/Notifications/NotificationSender.php:174]: Ha, thanks for pointing this out
I had some chance this time because you looked to the old errors report, please have a look to the "5.3.0-dev" one instead :)

**Comment by rodrigopedra** [comment]
Hi @williamdes , thanks for the quick review.

Besides the one for `Response@throw`, the additional errors weren't in the original #35673 .

I will wait for the maintainers to take a look on those first as to me it is tricky and could be a matter of opinion.

- The `Response@throw` documents a supported feature added to avoid a BC
- The `cache` helper one uses a `@param dynamic` syntax that is strange to me, but can be there for a reason
- The `event` helper docblock matches the `Illuminate\Contracts\Events\Dispatcher@dispatch` docblock, therefore I think we could copy the method signature and delegate that. But if someone rebound the `events` string in the container to something else it could lead to a BC. So I will leave the decision to the maintainers for now.

File: src/Illuminate/Notifications/NotificationSender.php
Comment (rodrigopedra) [reply src/Illuminate/Notifications/NotificationSender.php:174]: Now I see, thanks for the heads up

**Comment by GrahamCampbell** [comment]
We are not changing the mailer stuff yet. We marked it as void, however some people are still using the return value, and we have no good way of providing an alternative for their use case yet, so that's how we arrived having a void contract, but mixed in reality. The mailer component may be getting some refactoring in 9.x if we decide to switch to symfony mailer. The issue with the void-mixed stuff may be resolved properly, at that point.

**Comment by williamdes** [comment]
Hi @GrahamCampbell 
So to fix #35673 a PR with only phpdoc fixes is needed ?

**Comment by GrahamCampbell** [comment]
The cache and validation fixes look good, though. Please submit a PR fixing that on 6.x (if the code exists on 6.x, otherwise 8.x is fine).

**Comment by williamdes** [comment]
And for 5.x issues ?

**Comment by rodrigopedra** [comment]
Hey @GrahamCampbell , thanks for your review

Wouldn't reverting the return for  `Illuminate\Mail\PendingMail@send` make it good? The other fixes on the mailer docblocks are legitimate in my opinion (removing an extra semicolon and fixing `setGlobalToAndRemoveCcAndBcc` to conform with interface)

If you think so I just pushed a commit to revert the `send` return to my fork.

**Comment by williamdes** [comment]
> Hey @GrahamCampbell , thanks for your review
> 
> Wouldn't reverting the return for `Illuminate\Mail\PendingMail@send` make it good? The other fixes on the mailer docblocks are legitimate in my opinion (removing an extra semicolon and fixing `setGlobalToAndRemoveCcAndBcc` to conform with interface)
> 
> If you think so I just pushed a commit to revert the `send` return to my fork.

I think you can re-open a pull-request :)

**Comment by taylorotwell** [comment]
@rodrigopedra re-opened. Is this ready to go?

**Review by williamdes — APPROVED** [review]

**Comment by GrahamCampbell** [comment]
Some of these need fixing on the 6.x series, first.

**Comment by rodrigopedra** [comment]
Hi @taylorotwell I just force-pushed again to force re-syncing the PR.
 
So now there are just docblock changes. No code changes. I think it is good to go.

Again, I left the `Illuminate\Http\Client\Response@throw` as it is now, inconsistent but documenting a supported dynamic parameter.

I also didn't change the `cache` and `event` helpers docblocks.

I think the `event` one could be changed, please refer to comment https://github.com/laravel/framework/pull/35675#issuecomment-748537250 for more details.

I can send the needed fixes for 6.x too.

**Comment by rodrigopedra** [comment]
If you want I can make the changes from comment https://github.com/laravel/framework/pull/35675#issuecomment-748537250 before merging.

**Comment by williamdes** [comment]
By the way I just want to say that the variadic is not well documented
Sorry for the spamming but it is important to say

Please refer to https://github.com/20Tauri/DoxyDoxygen/issues/135#issuecomment-512090231

**Pull request merged**
